internal/installer/elasticagent_tar_darwin.go

```diff
@@ -0,0 +1,184 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package installer
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/elastic/e2e-testing/internal/common"
+	"github.com/elastic/e2e-testing/internal/deploy"
+	"github.com/elastic/e2e-testing/internal/kibana"
+	"github.com/elastic/e2e-testing/internal/utils"
+	log "github.com/sirupsen/logrus"
+	"go.elastic.co/apm"
+)
+
+// elasticAgentTARDarwinPackage implements operations for a TAR installer
+type elasticAgentTARDarwinPackage struct {
+	service deploy.ServiceRequest
+	deploy  deploy.Deployment
+}
+
+// AttachElasticAgentTARDarwinPackage creates an instance for the TAR installer
+func AttachElasticAgentTARDarwinPackage(deploy deploy.Deployment, service deploy.ServiceRequest) deploy.ServiceOperator {
+	return &elasticAgentTARDarwinPackage{
+		service: service,
+		deploy:  deploy,
+	}
+}
+
+// AddFiles will add files into the service environment, default destination is /
+func (i *elasticAgentTARDarwinPackage) AddFiles(ctx context.Context, files []string) error {
+	return nil
+}
+
+// Inspect returns info on package
+func (i *elasticAgentTARDarwinPackage) Inspect() (deploy.ServiceOperatorManifest, error) {
+	return deploy.ServiceOperatorManifest{
+		WorkDir:    "/opt/Elastic/Agent",
+		CommitFile: "/elastic-agent/.elastic-agent.active.commit",
+	}, nil
+}
+
+// Install installs a package
+func (i *elasticAgentTARDarwinPackage) Install(ctx context.Context) error {
+	log.Trace("No TAR install instructions")
+	return nil
+}
+
+// Exec will execute a command within the service environment
+func (i *elasticAgentTARDarwinPackage) Exec(ctx context.Context, args []string) (string, error) {
+	span, _ := apm.StartSpanOptions(ctx, "Executing Elastic Agent command", "elastic-agent.tar.exec", apm.SpanOptions{
+		Parent: apm.SpanFromContext(ctx).TraceContext(),
+	})
+	span.Context.SetLabel("arguments", args)
```

Comment:
Can we add a label to set the GOOS? Basically to distinguish from the tar installer on Linux

Reply:
I'd add that label to all methods here, or to name the span with "tar_darwin"